engine/src/execution_graph/logic_controllers/BatchAggregationProcessing.h

```diff
@@ -284,24 +304,62 @@ class MergeAggregateKernel : public kernel {
 	}
 
 	virtual kstatus run() {
+        
+        kstatus status = kstatus::stop;
         CodeTimer timer;
-        CodeTimer eventTimer(false);
-
-        std::vector<std::unique_ptr<ral::frame::BlazingTable>> tablesToConcat;
-		std::vector<ral::frame::BlazingTableView> tableViewsToConcat;
-
+        
+        std::vector<std::unique_ptr<ral::cache::CacheData>> batches;
+		
         // This Kernel needs all of the input before it can do any output. So lets wait until all the input is available
         this->input_cache()->wait_until_finished();
 
         bool ordered = false; // If we start using sort based aggregations this may need to change
-        BatchSequence input(this->input_cache(), this, ordered);
-        int batch_count=0;
+        BatchSequenceBypass input(this->input_cache(), this);
         try {
             while (input.wait_for_next()) {
-                auto batch = input.next();
-                // std::cout<<"MergeAggregateKernel batch "<<batch_count<<std::endl;
-                // ral::utilities::print_blazing_table_view_schema(batch->toBlazingTableView(), "MergeAggregateKernel_batch" + std::to_string(batch_count));
-                batch_count++;
```

Comment:
how are you keeping track of batch_count now?

Reply:
this run function only has one batch, so its hardcoded to 0 in line 325

Reply:
well... let me refrase that. It only has one execution batch. It taked in several batches and concatenates them. But the concept of batch here should be execution batch